sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/util/RebaseDateTimeSuite.scala

```diff
@@ -409,4 +409,40 @@ class RebaseDateTimeSuite extends SparkFunSuite with Matchers with SQLHelper {
       }
     }
   }
+
+  test("SPARK-31959: JST -> HKT at Asia/Hong_Kong in 1945") {
```

Comment:
Please hold on this. This test case is flaky in `master` branch already.
- https://amplab.cs.berkeley.edu/jenkins/view/Spark%20QA%20Test%20(Dashboard)/job/spark-master-test-maven-hadoop-3.2-hive-2.3/lastCompletedBuild/testReport/org.apache.spark.sql.catalyst.util/RebaseDateTimeSuite/SPARK_31959__JST____HKT_at_Asia_Hong_Kong_in_1945/
```
- SPARK-31959: JST -> HKT at Asia/Hong_Kong in 1945 *** FAILED ***
  -766744200000000 did not equal -766747800000000 (RebaseDateTimeSuite.scala:432)
```